### PR TITLE
[Bullet] SDF features (bullet developing)

### PR DESCRIPTION
**Pull request by Lobotuerk**
Basic SDF features to allow the first simulation run, iterative work on the devel_bullet branch
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

**Review by Blast545 — APPROVED** [review]
I tested the branch locally and the gazebo simulator checks properly the features added. We might need to add tests later for this. LGTM to merge

File: bullet/src/SDFFeatures.cc [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+Identity SDFFeatures::ConstructSdfWorld(
+    const Identity &_engine,
+    const ::sdf::World &_sdfWorld)
+{
+  const Identity worldID = this->ConstructEmptyWorld(_engine, _sdfWorld.Name());
+
+  const WorldInfoPtr &worldInfo = this->worlds.at(worldID);
+
+  auto gravity = _sdfWorld.Gravity();
+  worldInfo->world->setGravity(btVector3(gravity[0], gravity[1], gravity[2]));
```
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc:32]: convertVec goes from eigen vector to bullet vector. I think _sdfWorld.Gravity gives back a std::vector, so dont think so

File: bullet/src/SDFFeatures.cc [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  const std::string name = _sdfModel.Name();
+  const auto pose = _sdfModel.RawPose();
+  const bool isStatic = _sdfModel.Static();
+  const bool selfCollide = _sdfModel.SelfCollide();
+
+  // Set multibody params
+  const btScalar baseMass = 0;
+  // TODO(lobotuerk) update mass calculation to something like
+  // https://github.com/bulletphysics/bullet3/blob/master/examples/MultiBody/MultiDofDemo.cpp#L289
+  const btVector3 baseInertiaDiag(0, 0, 0); // This has to be tested, Im not sure if 0 0 0 is normal
```
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc:51]: Even after the mass calculation the inertia vector was still 0 0 0

**Comment by Blast545** [comment]
I just noticed that the methods `convertMat` and `convertVec` are used through various of the features in the base bullet branch, can we move them to the `Base.hh` file to reuse them conveniently?

File: bullet/src/SDFFeatures.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc:51]: You can check the link I left there

File: bullet/src/SDFFeatures.cc [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+Identity SDFFeatures::ConstructSdfLink(
+  const Identity &_modelID,
+  const ::sdf::Link &_sdfLink)
+{
+  // Read sdf params
+  const std::string name = _sdfLink.Name();
+  const auto pose = _sdfLink.RawPose();
+  const auto inertial = _sdfLink.Inertial();
+  const auto mass = inertial.MassMatrix().Mass();
+  const auto diagonalMoments = inertial.MassMatrix().DiagonalMoments();
```
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc:111]: By the base do you mean `base.hh`?

File: bullet/src/SDFFeatures.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (Blast545) [reply bullet/src/SDFFeatures.cc:51]: Just saw it, yeah, we can check this later again if needed. `0 0 0` doesn't seem to be an incorrect value there

File: bullet/src/SDFFeatures.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (Blast545) [reply bullet/src/SDFFeatures.cc:111]: By base I was talking to line 53, `const btVector3 baseInertiaDiag(0, 0, 0);` just to check if there wasn't something similar for that issue

**Comment by Lobotuerk** [comment]
Merging

**Pull request merged**